pkg/compactor/blocks_cleaner.go

```diff
@@ -449,6 +461,24 @@ func (c *BlocksCleaner) cleanUserPartialBlocks(ctx context.Context, partials map
 		level.Info(userLogger).Log("msg", "deleted partial block marked for deletion", "block", blockID)
 		return nil
 	})
+
+	// Check if partial block is older than delay period, and mark for deletion
+	for _, blockID := range partialsToMark {
+		if !partialDeletionCutoffTime.IsZero() {
+			lastModified, err := findLastModifiedTimeForBlock(ctx, blockID, userBucket, userLogger)
+			if err != nil {
+				level.Warn(userLogger).Log("msg", "failed to find last modified time for partial block", "block", blockID)
+			}
+			if !lastModified.IsZero() && lastModified.Before(partialDeletionCutoffTime) {
+				level.Info(userLogger).Log("msg", "stale partial block found: marking block for deletion", "block", blockID, "last modified", lastModified)
+				if err := block.MarkForDeletion(ctx, userLogger, userBucket, blockID, "stale partial block", c.blocksMarkedForDeletion); err != nil {
```

Comment:
Passing `c.blocksMarkedForDeletion` to `MarkForDeletion` is a bug. We can pass `c.partialBlocksMarkedForDeletion` directly, and not increment it in `else` clause.